src/xGadgetron/cGadgetron/include/sirf/Gadgetron/gadgetron_data_containers.h

```diff
@@ -514,8 +566,11 @@ namespace sirf {
 
 		virtual gadgetron::shared_ptr<ISMRMRD::Acquisition>
 			get_acquisition_sptr(unsigned int num) = 0;
-		virtual int get_acquisition(unsigned int num, ISMRMRD::Acquisition& acq) const = 0;
-		virtual void set_acquisition(unsigned int num, ISMRMRD::Acquisition& acq) = 0;
+		virtual int get_acquisition(unsigned int,
```

Comment:
in `get_data` and `set_data` you pass `ignore_mask` to `get_acquisitions` but not anywhere else. Should it not be done everywhere where `get_acquisitions` is called? 

An alternative could be to change the code such that the ignore mask is used when reading in the data from file but afterwards get_acquisition always returns everything in AcquisitionsData and the flags are disregarded. If the users wants a subset for a particular set of flags they can use `get_subset`.

Reply:
I like the subset idea...

Reply:
> in get_data and set_data you pass ignore_mask to get_acquisitions but not anywhere else. Should it not be done everywhere where get_acquisitions is called?

I would rather not have it in algebraic methods lest MR algebra would require separate interface as we do not use masks on PET and Reg sides.

> An alternative could be to change the code such that the ignore mask is used when reading in the data from file but afterwards get_acquisition always returns everything in AcquisitionsData and the flags are disregarded.

Some acquisition data include noise calibration data, which we do not use at present but might need later, but it should be ignored by algebraic operations.

> If the users wants a subset for a particular set of flags they can use `get_subset`.

In the presence of calibration data, the number of acquisitions in the subset of data needed for reconstruction might be just 1 acquisition less than the whole data, so creating a subset with `get_subset` would essentially duplicate a huge amount of data.

Reply:
The problem I have is that I have data with say 1000 acquisitions. 

If I use `pMR.AcquisitionData(praw+fdat, False, ignored = pMR.IgnoreMask(19))` then they are all read in correctly.

If I then call `print('Py dimensions ', acq_data.dimensions())` I get something like (500, 6, 128). I.e. 500 acquisitions are missed. This is because when `dimensions()` is called the algorithm goes through all acquisitions with `get_acquisition(i, acq)` to find out if the acquisition is valid. `get_acquisition` uses the default IgnoreMask() which is different to `IgnoreMask(19)` and hence ignores 500 acquisitions.

Reply:
@ckolbPTB it looks like you wanted to ignore acquisitions with flag `ISMRMRD_ACQ_IS_NOISE_MEASUREMENT = 19`, did you not? If so, then I have to apologise for not documenting `IgnoreMask` object properly, relying instead on self-explanatory names of variables and methods.

To ignore flag 19 one needs two steps:
```
ignored = IgnoreMask(0) # nothing to ignore - all 64 bits are zeros
ignored.ignore(19) # ignore acquisitions with flag ISMRMRD_ACQ_IS_NOISE_MEASUREMENT (nonzero bit 19)
```

Reply:
I changed my code based on your suggestions but it still does not work. I still only get 500 acquisitions instead of 1000.

To me I always see a problem if a different mask is used when reading in the data compared to calling `get_acquisition(i, acq)` later in the code. I understand the problem with the noise measurements because they need to be ignored for any image reconstruction process later, but there are lots of other acquisition flags which the user would want to have in the reconstruction pipeline and they are not covered by the default `IgnoreMask` used in  `get_acquisition(i, acq)`. I think we need to discuss this in more detail in the next software meeting.

Reply:
It looks like my suggestion was a bit too laconic, I should have added the third line:
```
ignored = IgnoreMask(0) # nothing to ignore - all 64 bits are zeros
ignored.ignore(19) # ignore acquisitions with flag ISMRMRD_ACQ_IS_NOISE_MEASUREMENT (nonzero bit 19)
pMR.AcquisitionData(praw+fdat, False, ignored=ignored) # use your handmade mask to ignore noise measurement acquisition only
```
or perhaps just one line `pMR.AcquisitionData(praw+fdat, False, ignored=pMR.IgnoreMask(0).ignore(19))` would work too.

What I mislooked though was that all this would have no effect on the dimensions because at present `AcquisitionData.dimensions` does not get ignored mask - I need to add it to the argument list (and in a few other places too). 

Furthermore, I started to suspect that I might have misunderstood the whole idea of these flags. The only flag that may cause ignoring an acquisition that I understand is  `ISMRMRD_ACQ_IS_NOISE_MEASUREMENT`. It occurred to me just now that e.g. flags  `ISMRMRD_ACQ_USER1` etc. that in my current design would cause ignoring the acquisition actually should not have such effect, serving some other purposes.

Reply:
@ckolbPTB as agreed on last Thu tcon, I changed the default value of `IgnoreMask.mask` to `1 << 18` - please check if this solves the problems you reported, so I can merge this PR.

Reply:
It does work now for Paul's data but I don't think this solution will work for data in general. The main problem I see is that we are using different IgnoreMasks throughout the code. 

When reading AcquisitionData from file we can specify an IgnoreMask. That is fine. 
Later in the code when e.g. get_acquisition_dimension() is called, get_acquisition() is called (https://github.com/SyneRBI/SIRF/blob/b6bdc997df50978807175648c2145d2141810d9e/src/xGadgetron/cGadgetron/gadgetron_data_containers.cpp#L192C14-L192C29) without specifying an IgnoreMask. This means get_acquisitions() uses the default IgnoreMask. If the user has specified a specific IgnoreMask when reading in the data (e.g. ignoring also some navigator data in addition to the nosie data) this is not used here and the code will probably crash. 

The only way I can see is to:
- make IgnoreMask an attribute of the AcquisitionData class
- allow the user to modify IgnoreMask to adapt it to the current steps in the reconstruction pipeline
- use AcquisitionData.IgnoreMask in e.g. get_acquisition()

Reply:
@ckolbPTB thanks, implemented